shell/packages/accounts-email-token/token_server.js

```diff
@@ -81,15 +90,34 @@ Accounts.registerLoginHandler("email", function (options) {
     };
   }
 
-  if (!consumeToken(user, options.token.trim())) {
+  const tokenString = options.token.trim();
+  const maybeToken = consumeToken(user, tokenString);
+  if (!maybeToken) {
     console.error("Token not found:", options.email);
     return {
       error: new Meteor.Error(403, "Invalid authentication code"),
     };
   }
 
+  // Attempt to decrypt the resumePath, if provided.
+  let resumePath = undefined;
+  if (maybeToken.secureBox) {
+    const box = maybeToken.secureBox;
+    if (box.version === 1) {
+      const key = crypto.pbkdf2Sync(tokenString, box.salt, V1_ROUNDS, V1_KEYSIZE, V1_HASHFUNC);
```

Comment:
Hmm.  It's difficult to tell for sure - pbkdf2 uses an HMAC based on the given hash algorithm, but then HMAC-SHA256 will either pad the input with zeros if shorter than the block size, or...hash it if it's longer than the block size.

Perhaps we're better off switching V1_HASHFUNC to sha512 instead?